src/client/Statistics.ts

```diff
@@ -20,7 +20,50 @@ import {
  * first would be .2*2=.4
  */
 export function calculateAverages(responseData: GetFullScoresResponseData): SubmissionAverageStatistics {
-  throw new Error("Not Yet Implemented");
+  let cloned = [...responseData.assignments];
+  let subIndices: Array<number | undefined> = [];
+
+  //The for loop iterates through the assignments array
+  //Then it checks if the type is 1 and it is not active
+  //Then if subIndices array doesnt contain the respective submission Index, then add it to subIndices array
+  for (var i = 0; i < cloned.length; i++) {
+    if (cloned[i].type === 1 && !(cloned[i].active)) {
```

Comment:
You should import and use the `ASSIGNMENT_TYPE_RATING` constant in `constants.ts`.